### PR TITLE
feat(modules): add bank transient store, endblocker callback

### PR DESCRIPTION
**Pull request by vinhphuctadang**
## Goal:
- Get all updated balances once in endblock, indexer just need to read it
- Fluxd control the event emission logic, chainstream compatible 

## Changes

- Add transient store for added metadata, updated balances
- Add callback in endblocker so that logic could be handled on fluxd 

I think it's neat enough to be maintained in future cosmos sdk version if we want to migrate

### CODE REVIEW

File: x/bank/keeper/end_blocker.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package keeper
+
+import (
+	"context"
+)
+
+func (k BaseKeeper) EndBlocker(ctx context.Context) error {
+	if k.endBlockerCb != nil {
+		return k.endBlockerCb(&k, ctx)
```
Comment (nhannamsiu) [inline x/bank/keeper/end_blocker.go]: @vinhphuctadang let's make it generic and accessible to all modules by adding switch case.

Also add specific names for callbacks we support using map, like k.callbacks["module-name"] = interface{}.

File: x/bank/keeper/keeper.go [hunk 2]
```diff
@@ -45,22 +47,27 @@ type Keeper interface {
...
 type BaseKeeper struct {
 	BaseSendKeeper
 
 	ak                     types.AccountKeeper
 	cdc                    codec.BinaryCodec
 	storeService           store.KVStoreService
+	tStoreKey              *storetypes.TransientStoreKey
 	mintCoinsRestrictionFn types.MintingRestrictionFn
 	logger                 log.Logger
+	endBlockerCb           func(*BaseKeeper, context.Context) error
```
Comment (nhannamsiu) [inline x/bank/keeper/keeper.go]: update to map of modules name to callback methods, if function footprints are similar then we can keep it, else we can do map[moduleName]interfaces{} and cast it to concrete methods later

File: x/bank/types/keys.go [hunk 3]
```diff
@@ -17,6 +17,8 @@ const (
 
 	// RouterKey defines the module's message routing key
 	RouterKey = ModuleName
+
+	TransientStoreKey = "transient_bank"
```
Comment (nhannamsiu) [inline x/bank/types/keys.go]: Should use `ModuleName` with short prefix like `"t_" + ModuleName`

**Comment by vinhphuctadang** [comment]
Retried with fluxd, looks good, also have expected events

image

**Review by nhannamsiu — APPROVED** [review]

**Pull request merged**